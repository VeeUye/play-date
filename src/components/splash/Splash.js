import React from "react";
import Title from "../atoms/Title";
import styles from "./splash.module.css";

const Splash = () => {
  return (
    <div className={styles.background}>
      <Title />
    </div>
  );
};

export default Splash;
