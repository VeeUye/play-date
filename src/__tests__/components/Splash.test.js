import React from "react";
import { render, screen } from "@testing-library/react";
import Splash from "../../components/splash/Splash";

describe("Splash", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Splash />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the title", () => {
    render(<Splash />);

    expect(screen.getByText(/Playdate/).textContent).toBe("Playdate");
  });

  it("displays the tagline", () => {
    render(<Splash />);

    expect(screen.getByText(/Scheduler/i).textContent).toBe(
      /scheduler for parents/i
    );
  });

  it("displays the splash image", () => {
    render(<Splash />);

    const image = screen.getByRole("img");

    expect(image).toHaveAttribute("src", "/some-image-path.png");
    expect(image).toHaveAttribute("alt", "some alt text");
    expect(image).toBeVisible();
  });

  it("displays a sign in button", () => {
    render(<Splash />);

    const signInButton = screen.getByRole("button", { name: /submit/i });

    expect(signInButton).toHaveTextContent(/Submit/i);
  });

  it("displays a sign up button", () => {
    render(<Splash />);

    const signUpButton = screen.getByRole("button", { name: /sign in/i });

    expect(signUpButton).toHaveTextContent(/sign up/i);
  });
});
