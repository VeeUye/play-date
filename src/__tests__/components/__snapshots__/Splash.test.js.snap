// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash renders correctly 1`] = `
<DocumentFragment>
  <div
    class="background"
  >
    <div
      class="splash__title"
    >
      Playdate
    </div>
    <div
      class="text"
    >
      Scheduler for parents
    </div>
    <img
      alt="girl on a swing"
      src="swinging.svg"
    />
    <button
      class="signIn"
    >
      Sign In
    </button>
    <button
      class="signUp"
    >
      Sign Up
    </button>
  </div>
</DocumentFragment>
`;
